### ISSUE TITLE
[feature request] visualisation over time

### ISSUE THREAD

**Issue by foobar0112** [paperscape/paperscape-mapclient#7]
It would interesting to see the landscape evolving over time. Is there already any approach like this?
One possible solution might be to just use the pre-computed maps, if they got stored… (?)
Or creating an animation with maps as "frames", which are generated ignoring the lest N moths or so…

**Comment by rknegjens** [comment]
That would indeed be cool. We do store the map locations, so in principle it's possible. In the past we've played with showing the map for different years. One problem is that we use tiles to render the map, so one (static) map already uses lots of diskspace. Probably the way to go is to dynamically render the map client side for high zoom levels, and only generate map tiles for low zoom e.g. when > 10k papers are showing at once.

**Comment by foobar0112** [comment]
That sound's reasonable. Initially it might be sufficient to render a non-interactive animation on a low overview-like zoom level, I think. 

I would like to do that. Can you provide this data? Or do I find all the required code in the repo(s) to render the maps on suitable zoom lvl on my own? 

Using your stored history of map locations (Do you mean the location like the "geo"-position of every article?) would result in an visualisation of 2013 until now, as far as I understand. But filtering the articles by there date of publication and generating maps for e.g. every year would be interesting as well.

**Comment by rknegjens** [comment]
You're right that we've only stored daily map locations from 2013 onwards. When we generated maps for each year (1991 to present) we ran a separate map generation for each year e.g. 1991 - 2007 for the 2007 map etc.

In principle the data and code to do this is available online. The map generation software is available in `paperscape-backend` repo, and the citation data in `paperscape-data` in csv format.

**Comment by sdbadger** [comment]
Hello. Interesting to see this was already discussed - I was just looking into the same possibility. Unfortunately my coding skills are not great and I had some compilation issues related to mysql. In case there were any developments regarding time evolution maps let me know!
